core/src/main/resources/error/error-classes.json

```diff
@@ -32,20 +44,36 @@
   "GROUPING_SIZE_LIMIT_EXCEEDED" : {
     "message" : [ "Grouping sets size cannot be greater than %s" ]
   },
+  "GROUP_INDEX_EXCEED_GROUP_COUNT" : {
+    "message" : [ "Regex group count is %s, but the specified group index is %s" ],
+    "sqlState" : "07009"
+  },
+  "GROUP_INDEX_LESS_THAN_ZERO" : {
+    "message" : [ "The specified group index cannot be less than zero" ],
+    "sqlState" : "07009"
+  },
   "IF_PARTITION_NOT_EXISTS_UNSUPPORTED" : {
     "message" : [ "Cannot write, IF NOT EXISTS is not supported for table: %s" ]
   },
   "INCOMPARABLE_PIVOT_COLUMN" : {
     "message" : [ "Invalid pivot column '%s'. Pivot columns must be comparable." ],
     "sqlState" : "42000"
   },
+  "INCOMPARABLE_TYPE" : {
```

Comment:
I think this should be more specific; eg. `CANNOT_GENERATE_CODE_FOR_UNCOMPARABLE_TYPE`.